### PR TITLE
Display default values in review dropdown

### PR DESCRIPTION
**Pull request by jmkulwik**
This adds a `ReviewNavigator` element that decides what should display in the review dropdown area. The ReviewNavigator is mostly a refactor of existing code, but it adds a few key elements:
- A default value if there are no peers to display
- An error message if we reach an error state in loading the peers

The error state

View when there are no other peers

The display with peers or from another student's project remains unchanged

## Links

- jira ticket: [CSA-704](https://codedotorg.atlassian.net/browse/CSA-704)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/src/templates/instructions/codeReview/ReviewNavigator.jsx [hunk 1]
```diff
@@ -0,0 +1,86 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+import DropdownButton from '@cdo/apps/templates/DropdownButton';
+import Button from '@cdo/apps/templates/Button';
+import javalabMsg from '@cdo/javalab/locale';
+
+class ReviewNavigator extends Component {
```
Comment (sanchitmalhotra126) [inline apps/src/templates/instructions/codeReview/ReviewNavigator.jsx:7]: Nice refactor 👍

File: apps/src/templates/instructions/codeReview/ReviewNavigator.jsx [hunk 2]
```diff
@@ -0,0 +1,86 @@
...
+    peers: PropTypes.array,
+    onSelectPeer: PropTypes.func,
+    onReturnToProject: PropTypes.func,
+    viewPeerList: PropTypes.bool,
+    loadError: PropTypes.bool
+  };
+
+  getPeerList() {
+    const {loadError, peers, onSelectPeer} = this.props;
+    if (loadError || !Array.isArray(peers)) {
```
Comment (sanchitmalhotra126) [inline apps/src/templates/instructions/codeReview/ReviewNavigator.jsx:18]: Won't react also throw an error if `peers` is not an array?

File: apps/i18n/javalab/en_us.json [hunk 3]
```diff
@@ -32,6 +32,7 @@
   "enablePeerReview": "Enable Peer Review",
   "errorJavabuilderConnectionGeneral": "We hit an error connecting to our server. Try again.",
   "errorJavabuilderConnectionNotAuthorized": "You are not authorized to access Javalab. Do you need to log in or join a section?",
+  "errorLoadingClassmates": "There was an error loading the list of your classmates. Switch to 'Instructions' and back to 'Review' to reload the list.",
```
Comment (sanchitmalhotra126) [inline apps/i18n/javalab/en_us.json:35]: Just checking, this will change soon once we can refresh on opening the dropdown right?

**Review by sanchitmalhotra126 — APPROVED** [review]
A couple questions but otherwise LGTM!

File: apps/src/templates/instructions/codeReview/ReviewNavigator.jsx (hunk 2 above)
Comment (jmkulwik) [reply apps/src/templates/instructions/codeReview/ReviewNavigator.jsx:18]: Tested this out to make sure:
* I was originally intending to check whether `peers` is undefined, which is a valid value for this prop. When it is undefined, this check returns false and we see the expected error state.
* When I passed in another type for peers (i.e. an object) react raised a warning but still displayed the component. This check returned false and we again see the expected error state.
* I then tried passing in an object for `peers` and disabling this check. The code instead hit an error when trying to access peers.map. This led to 7 errors in the console, one of which tried to unmount the ProtectedStatefulDiv. This led to a rather fatal error where the entire window went blank.

File: apps/i18n/javalab/en_us.json (hunk 3 above)
Comment (jmkulwik) [reply apps/i18n/javalab/en_us.json:35]: Yes

**Pull request merged**